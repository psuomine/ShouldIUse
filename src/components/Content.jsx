import React, { Component } from 'react'
import styled from 'styled-components'
import PageTitle from 'components/PageTitle'
import SearchCard from 'components/SearchCard'
import Repository from 'components/repository/Repository'
import ErrorMessage from 'components/ErrorMessage'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 96px;
  align-items: center;
  width: 1400px;
  padding: 0 24px;
`

const SearchCardLayout = styled.div`
  max-width: 600px;
  width: 100%;
`

class Content extends Component {
  state = {
    name: '',
    owner: '',
  }

  handlePostForm = (name, owner) => this.setState({ name, owner })

  render() {
    const { name, owner, errors } = this.state
    return (
      <Container>
        <PageTitle title="Should i use the NPM library?" />
        <SearchCardLayout>
          <SearchCard handlePostForm={this.handlePostForm} />
        </SearchCardLayout>

        {name && <Repository name={name} owner={owner} />}
        {errors &&
          errors.graphQLErrors &&
          errors.graphQLErrors.map(err => (
            <ErrorMessage message={err.message} />
          ))}
      </Container>
    )
  }
}

export default Content
