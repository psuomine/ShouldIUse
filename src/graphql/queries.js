export const getRepository = `
	query Repository($name: String!, $owner: String!) {
		repository(name: $name, owner: $owner) {
			createdAt,
			isArchived,
			hasIssuesEnabled,
			description,
			openIssues: issues(states: [OPEN]) {totalCount},
			homepageUrl,
			name,
			nameWithOwner,
			watchers {totalCount},
			stargazers {totalCount},
			forkCount
		}
	}
`
