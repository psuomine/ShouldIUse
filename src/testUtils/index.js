import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import Theme from 'theme/Theme'

const customRender = (node, options) => render(<ThemeProvider theme={Theme}>{node}</ThemeProvider>, options)

// re-export everything
export * from 'react-testing-library'

// override render method
export { customRender as render }
