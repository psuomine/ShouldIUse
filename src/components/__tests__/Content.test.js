import React from 'react'
import { render, fireEvent, waitForElement } from 'testUtils'
import gql from 'graphql-tag'
import { MockedProvider } from 'react-apollo/test-utils'
import { getRepository } from 'graphql/queries'
import Content from '../Content'

const mocks = [
  {
    request: {
      query: gql(getRepository),
      variables: {
        name: 'lodash',
        owner: 'lodash',
      },
    },
    result: {
      data: {
        repository: {
          createdAt: '2012-04-07T04:11:46Z',
          isArchived: false,
          hasIssuesEnabled: true,
          description: 'A modern JavaScript utility library delivering modularity, performance, & extras.',
          openIssues: {
            totalCount: 1,
          },
          homepageUrl: 'https://lodash.com/',
          name: 'lodash',
          nameWithOwner: 'lodash/lodash',
          watchers: {
            totalCount: 862,
          },
          stargazers: {
            totalCount: 36478,
          },
          forkCount: 3780,
        },
      },
    },
  },
]

describe('<Content />', () => {
  it('Makes query call and shows data', async () => {
    const { getByTestId, getByPlaceholderText, container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Content />
      </MockedProvider>
    )

    const nameInput = getByPlaceholderText('Package name')
    const ownerInput = getByPlaceholderText('Package owner')
    fireEvent.change(nameInput, { target: { value: 'lodash' } })
    fireEvent.change(ownerInput, { target: { value: 'lodash' } })
    fireEvent.click(getByTestId('action-button'))

    // wait Apollo query to be ready
    const openIssuesTitle = await waitForElement(() => getByTestId('stat-container'))

    expect(openIssuesTitle).toHaveTextContent('Open issues')
    expect(openIssuesTitle).toHaveTextContent('Forks')
    expect(openIssuesTitle).toHaveTextContent('Starred')
    expect(openIssuesTitle).toHaveTextContent('Watchers')
    expect(getByTestId('action-button')).toHaveAttribute('disabled')
    expect(container.firstChild).toMatchSnapshot()
  })
})
