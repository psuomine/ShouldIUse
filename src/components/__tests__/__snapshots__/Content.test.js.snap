// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> Makes query call and shows data 1`] = `
<div
  class="sc-chPdSV TliYb"
>
  <h1
    class="sc-bdVaJa gvZhOw"
    color="text"
  >
    Should i use the NPM library?
  </h1>
  <div
    class="sc-kgoBCf jDcGq"
  >
    <div
      class="sc-htoDjs bJvByQ"
    >
      <div
        class="sc-dnqmqq fpDMrV"
      >
        <h3
          class="sc-htpNat ctLKUJ"
          color="text"
          data-testid="title"
        >
          Search the npm package
        </h3>
      </div>
      <form
        data-testid="form"
      >
        <div
          class="sc-gzVnrw cQsWNL"
        >
          <input
            class="sc-bxivhb iuSYkE"
            name="name"
            placeholder="Package name"
            value=""
          />
          <input
            class="sc-bxivhb iuSYkE"
            name="owner"
            placeholder="Package owner"
            value=""
          />
        </div>
        <button
          class="sc-bZQynM hxnWGE"
          data-testid="action-button"
          disabled=""
          type="submit"
        >
          <p
            class="sc-ifAKCX fMakYv"
            color="muted"
            font-weight="500"
          >
            SEARCH PACKAGE
          </p>
        </button>
      </form>
    </div>
  </div>
  <div
    class="sc-kAzzGY hNDmZp"
  >
    <h2
      class="sc-bwzfXH fcawr"
      color="text"
    >
      <a
        class="sc-gZMcBi cPCvXL"
        href="https://lodash.com/"
        target="_blank"
      >
        lodash
      </a>
    </h2>
    <p
      class="sc-ifAKCX jDDZCo"
      color="text"
      font-weight="400"
    >
      A modern JavaScript utility library delivering modularity, performance, & extras.
    </p>
    <div
      class="sc-cSHVUG dCaUhE"
      data-testid="stat-container"
    >
      <div
        class="sc-jTzLTM ezEtjS"
      >
        <div
          class="sc-jzJRlG ibjRax sc-VigVT hPkoPu"
        >
          <svg
            class="sc-gqjmRU dxluif"
            color="rgb(23,195,123)"
            data-testid="svg-icon"
            height="18"
            width="24"
          >
            <path
              d="M7.42449 14.5737L2.42002 9.56926C1.86238 9.01162 0.975871 9.01162 0.418231 9.56926C-0.13941 10.1269 -0.13941 11.0134 0.418231 11.571L6.40929 17.5621C6.96693 18.1197 7.86774 18.1197 8.42538 17.5621L23.5818 2.42002C24.1394 1.86238 24.1394 0.975871 23.5818 0.418231C23.0241 -0.13941 22.1376 -0.13941 21.58 0.418231L7.42449 14.5737Z"
              data-testid="svg-icon-path"
            />
          </svg>
        </div>
        <h1
          class="sc-bdVaJa sc-fjdhpX eUeDvZ"
          color="primary"
          data-testid="stat-value"
        >
          1
        </h1>
        <p
          class="sc-ifAKCX dajiYn"
          color="muted"
          data-testid="stat-title"
          font-weight="400"
        >
          Open issues
        </p>
      </div>
      <div
        class="sc-jTzLTM ezEtjS"
      >
        <div
          class="sc-jzJRlG ibjRax sc-VigVT hPkoPu"
        >
          <svg
            class="sc-gqjmRU dxluif"
            color="rgb(23,195,123)"
            data-testid="svg-icon"
            height="18"
            width="24"
          >
            <path
              d="M7.42449 14.5737L2.42002 9.56926C1.86238 9.01162 0.975871 9.01162 0.418231 9.56926C-0.13941 10.1269 -0.13941 11.0134 0.418231 11.571L6.40929 17.5621C6.96693 18.1197 7.86774 18.1197 8.42538 17.5621L23.5818 2.42002C24.1394 1.86238 24.1394 0.975871 23.5818 0.418231C23.0241 -0.13941 22.1376 -0.13941 21.58 0.418231L7.42449 14.5737Z"
              data-testid="svg-icon-path"
            />
          </svg>
        </div>
        <h1
          class="sc-bdVaJa sc-fjdhpX eUeDvZ"
          color="primary"
          data-testid="stat-value"
        >
          3.8k
        </h1>
        <p
          class="sc-ifAKCX dajiYn"
          color="muted"
          data-testid="stat-title"
          font-weight="400"
        >
          Forks
        </p>
      </div>
      <div
        class="sc-jTzLTM ezEtjS"
      >
        <div
          class="sc-jzJRlG ibjRax sc-VigVT hPkoPu"
        >
          <svg
            class="sc-gqjmRU dxluif"
            color="rgb(23,195,123)"
            data-testid="svg-icon"
            height="18"
            width="24"
          >
            <path
              d="M7.42449 14.5737L2.42002 9.56926C1.86238 9.01162 0.975871 9.01162 0.418231 9.56926C-0.13941 10.1269 -0.13941 11.0134 0.418231 11.571L6.40929 17.5621C6.96693 18.1197 7.86774 18.1197 8.42538 17.5621L23.5818 2.42002C24.1394 1.86238 24.1394 0.975871 23.5818 0.418231C23.0241 -0.13941 22.1376 -0.13941 21.58 0.418231L7.42449 14.5737Z"
              data-testid="svg-icon-path"
            />
          </svg>
        </div>
        <h1
          class="sc-bdVaJa sc-fjdhpX eUeDvZ"
          color="primary"
          data-testid="stat-value"
        >
          36.5k
        </h1>
        <p
          class="sc-ifAKCX dajiYn"
          color="muted"
          data-testid="stat-title"
          font-weight="400"
        >
          Starred
        </p>
      </div>
      <div
        class="sc-jTzLTM ezEtjS"
      >
        <div
          class="sc-jzJRlG ibjRax sc-VigVT hPkoPu"
        >
          <svg
            class="sc-gqjmRU dxluif"
            color="rgb(23,195,123)"
            data-testid="svg-icon"
            height="18"
            width="24"
          >
            <path
              d="M7.42449 14.5737L2.42002 9.56926C1.86238 9.01162 0.975871 9.01162 0.418231 9.56926C-0.13941 10.1269 -0.13941 11.0134 0.418231 11.571L6.40929 17.5621C6.96693 18.1197 7.86774 18.1197 8.42538 17.5621L23.5818 2.42002C24.1394 1.86238 24.1394 0.975871 23.5818 0.418231C23.0241 -0.13941 22.1376 -0.13941 21.58 0.418231L7.42449 14.5737Z"
              data-testid="svg-icon-path"
            />
          </svg>
        </div>
        <h1
          class="sc-bdVaJa sc-fjdhpX eUeDvZ"
          color="primary"
          data-testid="stat-value"
        >
          862
        </h1>
        <p
          class="sc-ifAKCX dajiYn"
          color="muted"
          data-testid="stat-title"
          font-weight="400"
        >
          Watchers
        </p>
      </div>
    </div>
  </div>
</div>
`;
